### ISSUE TITLE
[Bug] [iOS] Vibration.Vibrate not working in some case

### ISSUE THREAD

**Issue by Twelve57** [xamarin/Essentials#1080]
### Description
I've created and subscribed to an EventHandler to catch the beginning of an event (in my case the Audio Recorder with a Start to record method)
In this method callback, I call the Vibration.Vibrate method of Essentials to vibrate the device.
I've double checked the event callback method and also displayed a toast message in this method using the ACR.Userdialogs plugin and the toast is displayed as expected. 
But the vibration works only Android and not on my iOS devices.

### Steps to Reproduce

1. Create and Subscribe to an EventHandler to catch the beginning of an event
2. In this callback/event method call the Vibration.Vibrate method of the Essentials Plugin

### Expected Behavior

The vibration should works on Android **AND** iOS devices.

### Actual Behavior

The vibration works only on Android devices.

### Basic Information

- Version with issue: 1.3.1
- Last known good version: -
- IDE: VS for mac
- Platform Target Frameworks: 
  - iOS:  8.0
  - Android: 9.0
- Affected Devices:
iPhone 6 with 12.4.3
iPhone 7 Plus with 13.3

**Comment by jamesmontemagno** [comment]
Assumption here is you may need to invoke on main thread. Can you try that and see.

Also make sure that your device is set to vibrate on ring.

**Comment by Twelve57** [comment]
Yes I already tried that. The only difference it's the call from the Event Handler callback.
And yes, the devices are set to vibrate on ring.

Like I said, the vibration works for all of my pages, but not in a Event Handler method... really weird :x

**Comment by jamesmontemagno** [comment]
do you have a small sample we can test out? are you saying like in an event handler for a button click?

**Comment by mattleibow** [comment]
@Twelve57 did you ever manage to get this working?

**Comment by Twelve57** [comment]
Nope :/

**Comment by ignasbagdonas** [comment]
Have the same issue. Any solutions?

**Comment by jamesmontemagno** [comment]
do you have a small sample we can test out? are you saying like in an event handler for a button click?

**Comment by ignasbagdonas** [comment]
It's simple like that.

```
public override void ViewDidLoad()
{
    //Works
    Vibration.Vibrate();
    MyButton.TouchUpInside += OnButtonClick;
}

public void OnButtonClick(object sender, EventArgs e)
{
    //Not working
    Vibration.Vibrate();
   
    //Other stuff
}
```

**Comment by jamesmontemagno** [comment]
What if you call run on main thread in that OnButtonClick?

**Comment by ignasbagdonas** [comment]
Tried. Not working. 
Also tried this one:
           ```
 SystemSound.Vibrate.PlayAlertSound();
            MainThread.BeginInvokeOnMainThread(() =>
            {
                SystemSound.Vibrate.PlayAlertSound();
            });
```

Not working too in event handler. But works in view did load method. So, I don't found any issue with iOS, so I suspect that is Xamarin.iOS bug...

**Comment by jamesmontemagno** [comment]
interesting, yeah i think it isn't us at that point, very strange....

**Comment by mattleibow** [comment]
Maybe you can use the haptic feedback for this? Maybe newer versions of iOS need new permissions or something?

**Comment by ignasbagdonas** [comment]
I tried haptic feedback too and it not working also. My issue was that I'm using microphone at that time. It's camera view...

**Comment by jl-mobitech** [comment]
Any updates?
